Framework/DataHandling/src/LoadILLTOF2.cpp

```diff
@@ -214,8 +225,18 @@ void LoadILLTOF2::initWorkSpace(NeXus::NXEntry &entry,
   m_localWorkspace = WorkspaceFactory::Instance().create(
       "Workspace2D", m_numberOfHistograms + numberOfMonitors,
       m_numberOfChannels + 1, m_numberOfChannels);
-  m_localWorkspace->getAxis(0)->unit() = UnitFactory::Instance().create("TOF");
-  m_localWorkspace->setYUnitLabel("Counts");
+
+  NXClass monitor = entry.openNXGroup(m_monitorName);
+  if (monitor.containsDataSet("time_of_flight")) {
+    m_localWorkspace->getAxis(0)->unit() =
+        UnitFactory::Instance().create("TOF");
+    m_localWorkspace->setYUnitLabel("Counts");
+  } else {
+    g_log.debug("PANTHER diffraction mode");
+    m_localWorkspace->getAxis(0)->unit() =
+        UnitFactory::Instance().create("Wavelength");
```

Comment:
I think it's worth adding a note in the documentation, that first, it supports the diffraction mode, then second, for that mode the unit of the workspace will be wavelength and not time-of-flight.